src/Html/BlockBuilder.php

```diff
@@ -75,30 +75,21 @@ public function endBlock($append = false)
     }
 
     /**
-     * Sets a content of the layout block.
+     * Sets a content of the layout block, overwriting any previous content for that block.
      *
      * Output buffering is not used for this method.
-     *
-     * @param string $name Specifies the block name.
-     * @param string $content Specifies the block content.
-     * @return void
-     * @throws \Exception if there are no items in the block stack
      */
-    public function set($name, $content)
+    public function set(string $name, string $content): void
     {
         $this->blocks[$name] = $content;
     }
 
     /**
-     * Appends a content of the layout block.
+     * Appends content to a layout block.
      *
      * Output buffering is not used for this method.
-     *
-     * @param string $name Specifies the block name.
-     * @param string $content Specifies the block content.
-     * @return void
      */
-    public function append($name, $content)
+    public function append(string $name, string $content): void
     {
         if (!isset($this->blocks[$name])) {
             $this->blocks[$name] = '';
```

Comment:
Correct - if you try to append a block that doesn't exist, this acts more or less like the `set` method. The second time you run it, the block would exist and this would therefore append to the block.